nexenta-fusion-installer.sh

```diff
@@ -56,29 +61,47 @@ calculateRAM() {
         totalMemory=$(system_profiler SPHardwareDataType | grep "  Memory:" | grep -Eo '[0-9]+')
         # for mac users we expect that they all have RAM more than 1g and which can be divided by 2
         # in other cases we set 1g as default
-        if [[ $totalMemory -lt 2 ]]; then
-            isLowMemory=true
+        defaultHeapSize=$(( ${totalMemory} / 2 ))
+        
+        if [ $defaultHeapSize -gt $defaultHeapSizeLimitG ]; then
+            defaultHeapSize=$defaultHeapSizeLimitG
         fi
-        defaultHeapSize=$(( ${totalMemory} / 2 ))g
+
+        defaultHeapSize="${defaultHeapSize}g"
+
         if ! [[ $defaultHeapSize =~ ^[0-9]+g$ ]]; then
             defaultHeapSize="1g"
         fi
         totalMemory=${totalMemory}g
     elif [ -f "/proc/meminfo" ]; then
         # linux way to find total RAM
+        # total memory is presented in kb
         totalMemory=$(cat /proc/meminfo | grep -Po "(?<=MemTotal:)(\s+)(\w+)" | grep -Eo "\w+")
         # converting to bytes
-        # if total memory is lower than 2g
-        if [[ $totalMemory -lt 1024*1024*2 ]]; then
+        # if total memory is lower than 1gb
+        # we won't allow to run a container via this script
+        if [[ $totalMemory -lt 1000*1000 ]]; then
             isLowMemory=true
         fi
-        # total memory is presented in kib
-        totalMemory=$(( totalMemory * 1024 ))
+        # converting totalMemory to bytes
+        totalMemory=$(( totalMemory * 1000 ))
```

Comment:
I think it's better to convert total memory to gb and then perform a division by 2. In that case you would get truncated integer value (5/2 = 2)